docs/contrast.md

```diff
@@ -0,0 +1,212 @@
+# Contrast
+
+The color used for text and icons
+needs to be sufficiently different
+from the background color
+to be easily seen.
+
+Insufficient contrast affects reading speed,
+and measuring contrast is also important
+to ensure accessibility,
+including for those with low vision (Legge 1987):
+
+> People with low vision usually require large characters to read, so high contrast is particularly important for them.
+
+For people with normal vision,
+even when reading low-contrast text,
+increasing the character size makes it possible to maintain
+same maximum reading speed as when reading high-contrast text
+(Fujita _et. al._).
+
+## Luminance and chromatic contrast
+
+Contrast can be produced by sufficient difference in lightness _(luminance contrast)_,
+and also by sufficient difference in hue or chroma _(chromatic contrast)_.
+
+<img src="./images/luminance-contrast.png">
+
+<img src="./images/chromatic-contrast.png">
+
+Studies (Legge 1990) of both people with normal color vision,
+and people with low vision,
+have shown that sufficient luminance contrast
+is the primary factor affecting reading speed,
+and that adding chromatic contrast does not further increase reading speed.
+This is true for both those with low vision and those with normal vision.
+
+Color.js provides several methods to estimate luminance contrast.
+The strengths, drawbacks and uses of these methods have been examined by
+the NASA (Color Usage Research Lab).
+
+## Weber Contrast
+
+Weber contrast, also often simply called Luminance Contrast,
+is the relationship between the luminance Y<sub>max</sub>
+of a brighter area of interest
+and that of an adjacent darker area, Y<sub>min</sub>.
+It is commonly used in lighting design,
+and also to examine small, sharp edged symbols and text
+on larger, uniformly-colored backgrounds.
+
+> WC = Y<sub>max</sub> - Y<sub>min</sub> / Y<sub>min</sub>
+
+Any two colors can be used,
+and this formula does not care which is the text color and which is the background.
+The lighter of the two will be detected automatically.
+
+```js
+let color1 = new Color("p3", [0.9, 0.8, 0.1]);
+let color2 = new Color("slategrey");
+let contrast = contrastWeber(color1, color2);
```

Comment:
> Also, we use the OO API in examples, because that has better ergonomics.

Sure but, as I haven't written the OO API for those I was documenting what I have implemented, hence the draft PR.

Reply:
> or… (assuming we define a contrast() function/method.

We might, but 

- there isn't one
- the default, if any, needs to be defined

which is why I didn't document it (yet)